packages/engine/src/datastore/table/sync.rs

```diff
@@ -1,15 +1,91 @@
 use std::{fmt, sync::Arc};
 
+use futures::future::join_all;
 use parking_lot::RwLock;
 
 use crate::{
     datastore::{
         prelude::ContextBatch,
         table::pool::{agent::AgentPool, message::MessagePool},
     },
-    worker::runner::comms::inbound::InboundToRunnerMsgPayload,
+    simulation::comms::message::{SyncCompletionReceiver, SyncCompletionSender},
+    worker::{
+        error::{Error as WorkerError, Result as WorkerResult},
+        runner::comms::inbound::InboundToRunnerMsgPayload,
+    },
 };
 
+/// A state sync message with a tokio channel, so that
+/// the sender of the message can wait for the state
+/// sync to complete and find out whether the state
+/// sync succeeded.
+///
+/// Fields:
+/// `completion_sender`: After finishing or failing to sync state,
+///                      used by the receiver/handler of the message
+///                      to notify the sender of the message that the
+///                      state sync completed.
+/// `agent_pool`: Agent batches to load state from
+/// `message_pool`: Message batches to load state from
+#[derive(derive_new::new)]
```

Comment:
Chatted in Slack, we don't want these for structs with only pub fields, but can be in a follow-up PR.

```suggestion
// TODO: remove derive_new from structs with all pub fields
#[derive(derive_new::new)]
```